data-access-layer/bosh/BoshDirectorClient.js

```diff
@@ -542,16 +543,84 @@ class BoshDirectorClient extends HttpClient {
       if (this.deploymentIpsCache[deploymentName] !== undefined) {
         return this.deploymentIpsCache[deploymentName];
       } else {
-        return this
-          .getDeploymentInstances(deploymentName)
-          .reduce((ipList, instance) => ipList.concat(instance.ips), [])
-          .tap(response => {
-            logger.info(`Cached Ips for deployment - ${deploymentName} - `, response);
-            this.deploymentIpsCache[deploymentName] = response;
+        return this.getDeploymentIpsFromResource(deploymentName)
+          .then(ips => {
+            if (!_.isEmpty(ips)) {
+              logger.info(`[getDeploymentIps] Cached Ips for deployment - ${deploymentName} - `, ips);
+              this.deploymentIpsCache[deploymentName] = ips;
+              return ips;
+            } else {
+              return this.getDeploymentIpsFromDirector(deploymentName)
+                .then(ips => {
+                  logger.info(`[getDeploymentIps] Caching IPs on ApiServer and locally`);
+                  this.deploymentIpsCache[deploymentName] = ips;
+                  return this.putDeploymentIpsInResource(deploymentName, ips)
+                    .then(() => ips);
+                });
+            }
           });
       }
     });
+  }
+
+  parseServiceInstanceIdFromDeployment(deploymentName) {
```

Comment:
This function is present in [ServiceFabrikAdminController](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/api-controllers/ServiceFabrikAdminController.js#L143-L149) as well , better to keep it in utils ?

Reply:
Will move this function to utils. Initially I had more specific code under this, so made a copy here.